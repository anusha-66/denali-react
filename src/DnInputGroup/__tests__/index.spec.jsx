/**
 * Copyright 2020, Verizon Media
 * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
 */

import React from 'react';
import { render } from 'enzyme';
import { DnInputGroup, DnInput } from '../../index'; // src/index.tsx

describe('DnInputGroup Tests', () => {
    it('should render with default props', () => {
        expect.assertions(1);

        const rendered = render(
            <DnInputGroup>
                <DnInput />
            </DnInputGroup>,
        );

        expect(rendered).toMatchSnapshot();
    });

    it('should render correctly with custom props', () => {
        expect.assertions(1);

        const id = 'someId';
        const rendered = render(
            <div className="Foo">
                <DnInputGroup isStacked isResponsive isAuto labelText="Label" labelId={id}>
                    <DnInput id={id} />
                </DnInputGroup>
            </div>,
        );

        expect(rendered).toMatchSnapshot();
    });
});
