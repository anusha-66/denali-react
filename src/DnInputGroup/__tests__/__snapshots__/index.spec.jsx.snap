// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DnInputGroup Tests should render correctly with custom props 1`] = `
<div
  class="Foo"
>
  <div
    class="input-group is-stacked is-responsive auto"
  >
    <label
      for="someId"
    >
      Label
    </label>
    <span>
      <div
        class="input is-default"
      >
        <input
          id="someId"
          type="text"
        />
      </div>
    </span>
  </div>
</div>
`;

exports[`DnInputGroup Tests should render with default props 1`] = `
<div
  class="input-group"
>
  <span>
    <div
      class="input is-default"
    >
      <input
        type="text"
      />
    </div>
  </span>
</div>
`;
